src/Component/Product/ProductInterface.php

```diff
@@ -476,9 +476,9 @@ public function hasOneMainCategory();
     /**
      * Validates if product has one main category
      *
-     * @param ExecutionContext $context
+     * @param LegacyExecutionContext $context
      *
      * @return void
      */
-    public function validateOneMainCategory(ExecutionContext $context);
+    public function validateOneMainCategory(LegacyExecutionContext $context);
```

Comment:
I have modified this way:

public function validateOneMainCategory(ExecutionContextInterface $context)